archive/s/scala/reverse-string.scala

```diff
@@ -0,0 +1,26 @@
+import scala.io.StdIn.readLine
+
+object ReverseStringSample {
+  // revese using recursive & pattern matching
+  def reverseString(str: String): String = str.length match {
+    case 0 => ""
+    case 1 => str
+    case _ => reverseString(tail(str)) ++ head(str)
+  }
+
+  def head(str: String): String = str.length match {
+    case 0 => ""
+    case _ => str.slice(0, 1)
+  }
+
+  def tail(str: String): String = str.length match {
+    case 0 => ""
+    case 1 => ""
+    case _ => str.slice(1, str.length)
+  }
+
+  def main(args: Array[String]) {
+    val inputStr = readLine
```

Comment:
Right idea, but we want to read directly from args (rather than stdin). I don't know the syntax, but I'll take my best guess:

```suggestion
    val inputStr = args[0]
```

Of course, the solution should verify that there is something at 0 to grab: https://sample-programs.therenegadecoder.com/projects/reverse-a-string/

Reply:
Right, it'd be `args(0)` :smile:
Will be able to make the change shortly